### PR TITLE
feat: Add configurable end operation statuses plus skipped

### PR DESCRIPTION
**Pull request by jimmidyson**
This commit refactors the coloring of interactive output to use a library
to provide colors, plus introduces a new function `EndOperationWithStatus`
to allow for configurable output statuses whilst preserving backwards
compatibility. A new skipped status is provided ootb as well.

This change introduces a change in output: reset at the end of colored output
is done by using `\x1b[39m` escape code rather than `\x1b[0m`. As defined in
ANSI escape codes, 39 is used to return to previously defined color code rather
than a full reset, which 0 would provide.

Example output for skipped end operation status using a yellow color
if stderr is colored:

```
∅ skipped
```

### CODE REVIEW

File: core/output/interactive.go [hunk 1]
```diff
@@ -8,13 +8,8 @@ import (
...
 	"strings"
 	"sync"
-)
 
-const (
-	termRed    = "\x1b[31m"
-	termGreen  = "\x1b[32m"
-	termYellow = "\x1b[33m"
-	termReset  = "\x1b[0m"
+	"github.com/jwalton/gchalk"
```
Comment (mikolajb) [inline core/output/interactive.go:12]: 😎

File: core/output/output.go [hunk 2]
```diff
@@ -117,6 +117,20 @@ type Output interface {
 	//  output.EndOperation(true)
 	EndOperation(success bool)
```
Comment (mikolajb) [inline core/output/output.go:118]: Shall we add something like this:

```go
// Deprecated: EndOperation exists for historical compatibility
// and should not be used. To conclude the operation, use
// EndOperationWithStatus method.
```

**Review by mikolajb — APPROVED** [review]

File: core/output/output.go (hunk 2 above)
Comment (jimmidyson) [reply core/output/output.go:118]: Good point. On it.

Comment (jimmidyson) [reply core/output/output.go:121]: Added in d502a02e303db49c609db0b53fe48b0e4788fc47

**Pull request merged**